hack/make-rules/test-cmd-util.sh

```diff
@@ -1083,6 +1032,137 @@ run_kubectl_apply_tests() {
+  kube::test::get_object_assert 'pods test-pod' "{{${labels_field}.name}}" 'test-pod-label'
+  # Now, prune specifying new pods label, which does not prune previously created pod
+  kubectl apply --prune -l name=test-pod-label-2 -f hack/testdata/pod-2.yaml "${kube_flags[@]}"
+  kube::test::get_object_assert 'pods test-pod' "{{${labels_field}.name}}" 'test-pod-label'
+  kube::test::get_object_assert 'pods test-pod-2' "{{${labels_field}.name}}" 'test-pod-label-2'
+  # Clean up pods
+  kubectl delete pods test-pod "${kube_flags[@]}"
+  kubectl delete pods test-pod-2 "${kube_flags[@]}"
+
+  ## Test prune and dryRun does not really prune
+  # First, ensure that no pods exist
+  kube::test::get_object_assert pods "{{range.items}}{{$id_field}}:{{end}}" ''
+  # Next, create pod with apply, and ensure it exists
+  kubectl apply -f hack/testdata/pod.yaml "${kube_flags[@]}"
+  kube::test::get_object_assert 'pods test-pod' "{{${labels_field}.name}}" 'test-pod-label'
+  # Now, prune with dry-run does not actually create test-pod-2 or prune test-pod
+  kubectl apply --dry-run --prune --all -f hack/testdata/pod-2.yaml "${kube_flags[@]}"
+  kube::test::get_object_assert 'pods test-pod' "{{${labels_field}.name}}" 'test-pod-label'
+  output_message=$(! kubectl get pods test-pod-2 2>&1 "${kube_flags[@]}")
+  kube::test::if_has_string "${output_message}" 'pods "test-pod-2" not found'
+  # Clean up pods
+  kubectl delete pods test-pod "${kube_flags[@]}"
+
+  ## Non-namespaced resources get pruned (PersistentVolume) example
+  # First, ensure that no persistentvolumes exist
+  kube::test::get_object_assert persistentvolumes "{{range.items}}{{$id_field}}:{{end}}" ''
+  # Next, create persistentvolume through apply, and ensure it exists
+  kubectl apply -f hack/testdata/pv.yaml "${kube_flags[@]}"
+  kube::test::get_object_assert persistentvolumes "{{range.items}}{{$id_field}}:{{end}}" 'pv-test:'
+  # Next, apply a new peristentvolume and check the previous one was pruned
+  kubectl apply --prune --all -f hack/testdata/pv-2.yaml "${kube_flags[@]}"
+  kube::test::get_object_assert persistentvolumes "{{range.items}}{{$id_field}}:{{end}}" 'pv-test-2:'
+  kubectl delete persistentvolumes pv-test-2
+  
+  ## Test pruning only affects resources applied in "visited" namespaces.
+  kube::test::get_object_assert pods "{{range.items}}{{$id_field}}:{{end}}" ''
+  kubectl apply -f hack/testdata/pod.yaml
+  kube::test::get_object_assert 'pods test-pod' "{{${labels_field}.name}}" 'test-pod-label'
+  # Create pod in current namespace; pruning should not delete first pod
+  current_context=$(kubectl config current-context)
+  # Apply new pod into new namespace
+  namespace="apply-prune-test"
+  kube::log::status "Creating namespace $namespace"
+  kubectl create namespace "$namespace"
+  kubectl config set-context "apply-prune" --namespace="$namespace"
+  kubectl config use-context "apply-prune"
+  kubectl apply --prune --all -f hack/testdata/pod-2.yaml "${kube_flags[@]}"
```

Comment:
I have a feeling that we don't really need to change `context` during the test. Maybe just passing the `--namespace` flag here will work. Not sure does `kube_flags` contain here.

I could be wrong. If it won't work, let me know.

Reply:
Again, this is not changed (although I centralized the apply/prune tests into one location). In order to make this a low-risk PR, we'll keep the previous code unchanged, and we'll make fixes in future PR's.